sycl/test/linear_id/opencl-interop.cpp

```diff
@@ -1,3 +1,5 @@
+// REQUIRES: opencl
```

Comment:
I assume it means that this test is supposed to be skipped if there is no OpenCL support in the system.
If so, the code which detects presence of OpenCL looks strange:
```python
# PI API either supports OpenCL or CUDA.
opencl = False
if not cuda:
    opencl = True
    config.available_features.add('opencl')
```

Reply:
The current test idea is to either run testing for OpenCL xor CUDA.

(My understanding is though that some tests do not go through the SYCL default selector and therefore won't react to the env var - this adds some nondeterminism to which backend they choose).

The selection happens as follows:
* If the `check-sycl` target is called, then we want to test OpenCL.
* If the `check-sycl-cuda` target is called, then we want to test CUDA.
* By hand this would either be:
** `./bin/llvm-lit --param SYCL_BE=PI_OPENCL ./tools/sycl/test` or
** `./bin/llvm-lit --param SYCL_BE=PI_CUDA ./tools/sycl/test`

The [Python script for the LIT SYCL tests](https://github.com/intel/llvm/blob/sycl/sycl/test/lit.cfg.py#L88) uses the `SYCL_BE` environment variable to pass a backend parameter to the [`get_device_count_by_type`](https://github.com/intel/llvm/blob/sycl/sycl/tools/get_device_count_by_type.cpp) tool which only reports the available devices belonging to the parameter. It won't report CUDA devices when looking for OpenCL and vice versa.

Therefore, we either set the `cuda` xor the `opencl` feature but not both together.

However, it is always possible to explicitly query for all devices of all backends by not using the default device selector.

Reply:
I don't have neither OpenCL nor CUDA on machine. What will happen to my `check-sycl`?

Reply:
Just to clarify, I expect my tests to run on the host device only, but I'm not sure that tests which explicitly requested OpenCL will be disabled.

Reply:
Good question. I am not sure how to solve this cleanly right now, e.g., we would want to say that no backend is under test (to express that we test HOST)?

What will happen at the moment is:
* OpenCL testing is activated.
* No devices are found - so many test might skip while some fail.
* Some tests explicitly check that they are not on host before calling into OpenCL.

If I run `ninja check-sycl` on the tip of `sycl` on a machine without OpenCL but with CUDA I get the following:
```sh
Testing Time: 158.27s
********************
Failing Tests (5):
    SYCL :: kernel_from_file/hw.cpp   <- unexpected interaction with CUDA
    SYCL :: scheduler/DataMovement.cpp   <- unexpected interaction with CUDA
    SYCL :: scheduler/MultipleDevices.cpp   <- unexpected interaction with CUDA
    SYCL :: separate-compile/test.cpp    <- unexpected interaction with CUDA

  Expected Passes    : 201
  Unsupported Tests  : 17
  Unexpected Failures: 5
```

On a machine without OpenCL and when not building the CUDA backend no failures happen.

Something to look into though not in this PR I recon.

Reply:
NOTE: this file already has `// REQUIRES: opencl` at line 7.

Reply:
@bjoernknafla, I'm okay to resolve this separately. I need to go deeper into LIT config file to understand what is the best way to handle it.

Reply:
My bad, I missed the // REQUIRES: opencl at line 7. Let me check if I've missed any more.

Reply:
It's the only case.